services/cd-service/pkg/repository/transformer.go

```diff
@@ -3769,7 +3781,9 @@ func (c *envReleaseTrain) prognosis(
 		}
 
 		if state.DBHandler.ShouldUseOtherTables() {
-			release, err := state.DBHandler.DBSelectReleaseByVersion(ctx, transaction, appName, versionToDeploy, true)
+			release, exists := allLatestManifests[appName][versionToDeploy]
+			//release, err = state.DBHandler.DBSelectReleaseByVersion(ctx, transaction, appName, versionToDeploy, true)
+
 			if err != nil {
```

Comment:
this looks like it's never true

Reply:
Looks like it, but we have no guarantee that it isn't. The old code checks for release == nil, so this is the equivalent to that